### PR TITLE
Add Commands To GitLab Integration

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues And Description
As a part of this issue
https://github.com/demisto/etc/issues/37810
The implementation requires additional commands to be added to GitLab integration.
This PR is meant to add these commands to keep continuing the work on the mentioned ticket.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 1]
```diff
@@ -399,7 +399,207 @@ script:
...
+      type: String
+    - contextPath: GitLab.Job.stage
+      description: Stage of the job.
+      type: String
+    - contextPath: GitLab.Job.web_url
+      description: Web URL of the job.
+      type: String
+  - arguments:
+    - default: false
+      description: Project ID to retrieve pipeline schedules from.
```
Comment (dantavori) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:570]: fix description

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 2]
```diff
@@ -399,7 +399,207 @@ script:
...
+      required: true
+      secret: false
+    - default: false
+      description: ID of specific pipeline to retrieve its details.
+      isArray: false
+      name: pipeline_id
+      required: false
+      secret: false
+    deprecated: false
+    description: Get details of pipeline schedules.
```
Comment (dantavori) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:471]: fix description

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 3]
```diff
@@ -399,7 +399,207 @@ script:
...
+      description: Ref of the last pipe line that was run by the scheduled pipeline.
+        Relevant only when given pipeline schedule ID is given.
+      type: String
+    - contextPath: GitLab.PipelineSchedule.last_pipeline.status
+      description: Status of the last pipe line that was run by the scheduled pipeline.
+        Relevant only when given pipeline schedule ID is given.
+      type: String
+  - arguments:
+    - default: false
+      description: Project ID to retrieve pipeline schedules from.
```
Comment (dantavori) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:459]: fix description

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 4]
```diff
@@ -399,7 +399,207 @@ script:
...
+      type: Date
+    - contextPath: GitLab.Pipeline.duration
+      description: Duration of the pipeline.
+      type: Number
+    - contextPath: GitLab.Pipeline.web_url
+      description: Web URL of the pipeline.
+      type: String
+  - arguments:
+    - default: false
+      description: Project ID to retrieve pipeline schedules from.
```
Comment (dantavori) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:510]: fix description

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 5]
```diff
@@ -399,7 +399,207 @@ script:
...
+      required: true
+      secret: false
+    - default: false
+      description: ID of the pipeline to retrieve its jobs.
+      isArray: false
+      name: pipeline_id
+      required: true
+      secret: false
+    deprecated: false
+    description: Get details of pipeline schedules.
```
Comment (dantavori) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:522]: fix description

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 6]
```diff
@@ -241,6 +274,118 @@ def get_version_command(client, args):
...
+    job_id = args.get('job_id', '')
+    artifact_path_suffix = args.get('artifact_path_suffix', '')
+    response = client.get_job_artifact_request(project_id, job_id, artifact_path_suffix)
+    outputs = {
+        'job_id': job_id,
+        'artifact_path_suffix': artifact_path_suffix,
+        'artifact_data': response
+    }
+
+    return CommandResults(
```
Comment (dantavori) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:381]: please add a human readable

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 7]
```diff
@@ -241,6 +274,118 @@ def get_version_command(client, args):
...
+    response = client.get_pipeline_schedules_request(project_id, pipeline_schedule_id)
+    response = response if isinstance(response, list) else [response]
+    outputs = [{k: v for k, v in output.items() if k in PIPELINE_SCHEDULE_FIELDS_TO_EXTRACT} for output in response]
+
+    return CommandResults(
+        outputs_prefix='GitLab.PipelineSchedule',
+        outputs_key_field='id',
+        outputs=outputs,
+        raw_response=response,
+        readable_output=tableToMarkdown('GitLab Pipeline Schedules', outputs)
```
Comment (dantavori) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:300]: please use removeNull, headers and headerTransform args in all tableToMarkdown calls

**Review by dantavori — CHANGES_REQUESTED** [review]
great job! (🥁 )
see my minor notes, and don't forget to regenerate the readme

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py (hunk 6 above)
Comment (tomneeman151293) [reply Packs/DevSecOps/Integrations/GitLab/GitLab.py:381]: @dantavori About that, I added it a HR at first. But the artifact data can be quite large i guess (depending on which artifact retrieved). WDYT? I will add HR that the data for artifact and given job was retrieved if length is over 100, else output data as is

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py (hunk 7 above)
Comment (tomneeman151293) [reply Packs/DevSecOps/Integrations/GitLab/GitLab.py:300]: @dantavori How exactly do you want me to use headerTransform in the table to mark down?

**Review by dantavori — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@dantavori @ShahafBenYakir Doc review completed.  @tomneeman151293 Please regenerate the Integration README.

**Comment by shannon-holland** [comment]
@dantavori Doc review completed.

**Pull request merged**